src/kagglehub/models.py

```diff
@@ -13,7 +14,8 @@ def model_download(handle: str, path: Optional[str] = None):
     Returns:
         A string representing the path to the requested model files.
     """
-    return registry.resolver(handle, path)
+    h = parse_model_handle(handle)
```

Comment:
Instead of having each resolver call this method. Do it here instead and have the resolver take a ModelHandler instead of a string.